lightkurve/correctors/pldcorrector.py

```diff
@@ -82,199 +116,509 @@ class PLDCorrector(object):
-        # X has shape (n_components_first + n_components_higher_order + 1, n_cadences)
-        X = np.concatenate(X_sections, axis=1)
-
-        # set default transit mask
+        if design_matrix is None:
+            design_matrix = self.create_design_matrix(**kwargs)
         if cadence_mask is None:
-            cadence_mask = np.ones_like(lc.time, dtype=bool)
-        M = lambda x: x[cadence_mask[nanmask]]
-
-        # mask transits in design matrix
-        MX = M(X)
-
-        if use_gp:
-            # We use a Gaussian Process to model the long term trend.
-            # We do this by estimating the long term trend y by applying the
-            # preliminary PLD model defined above and subtracting it from the raw light curve.
-            # The "in transit" cadences are masked out in this step to prevent the
-            # long term approximation from over-fitting the transits.
-            XTX = np.dot(MX.T, MX)
-            XTX[np.diag_indices_from(XTX)] += 1e-8
-            XTy = np.dot(MX.T, M(rawflux))
-            y = M(rawflux) - np.dot(MX, np.linalg.solve(XTX, XTy))
-
-            # Estimate the amplitude parameter of a Matern-3/2 kernel GP
-            # by computing the standard deviation of y.
-            amp = np.nanstd(y)
-            tau = gp_timescale  # tau is a user-defined parameter
-            # set up gaussian process using celerite
-            # we use a Matern-3/2 kernel for its flexibility and non-periodicity
-            kernel = celerite.terms.Matern32Term(np.log(amp), np.log(tau))
-            gp = celerite.GP(kernel)
-            gp.compute(M(self.time), M(rawflux_err))
-
-            # compute the coefficients C on the basis vectors;
-            # the PLD design matrix will be dotted with C to solve for the noise model.
-            A = np.dot(MX.T, gp.apply_inverse(MX))
-            B = np.dot(MX.T, gp.apply_inverse(M(rawflux)[:, None])[:, 0])
+            cadence_mask = np.ones(len(self.raw_lc.time), dtype=bool)
+
+        # Theano raises an `AsTensorError` ("Cannot convert to TensorType")
+        # if the floats are not in 64-bit format, so we convert them here:
+        lc_time = np.asarray(self.raw_lc.time, np.float64)
+        lc_flux = np.asarray(self.raw_lc.flux, np.float64)
+        lc_flux_err = np.asarray(self.raw_lc.flux_err, np.float64)
+
+        # Covariance matrix diagonal
+        diag = lc_flux_err**2
+
+        # The cadence mask is applied by inflating the uncertainties in the covariance matrix;
+        # this is because celerite will run much faster if it is able to predict
+        # data for cadences that have been fed into the model.
+        diag[~cadence_mask] += 1e12
+
+        with pm.Model() as model:
+            # Create a Gaussian Process to model the long-term stellar variability
+            # log(sigma) is the amplitude of variability, estimated from the raw flux scatter
+            logsigma = pm.Normal("logsigma", mu=np.log(np.std(lc_flux)), sd=4)
```

Comment:
The sd is hard coded here...I think I'm happy with that but it would make a difficult error to debug if your priors were very off.